presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -92,7 +92,7 @@ public void run()
         boolean hasQuery = !Strings.isNullOrEmpty(clientOptions.execute);
         boolean isFromFile = !Strings.isNullOrEmpty(clientOptions.file);
 
-        if (!hasQuery || !isFromFile) {
+        if (!hasQuery && !isFromFile) {
```

Comment:
When stdout is redirected to some file then `AnsiOutputStream` is installed (since `isatty` will return `0` -- the output is not tty). With this fix we just bypass `AnsiConsole` if we know that we are in batch mode.